### PR TITLE
Bugfix tcp

### PR DESCRIPTION
**Pull request by boaks**
Correct parameter order in TcpServerConnector.
Correct "stop()" to be executable multiple times.

#118

### CODE REVIEW

**Comment by joemag1** [comment]
Kai, this looks good to me.

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/CoapTcpStack.java [hunk 1]
```diff
@@ -106,13 +107,15 @@ public CoapTcpStack(NetworkConfig config, Outbox outbox) {
 
 	// delegate to top
 	@Override public void sendRequest(Request request) {
+		if (null == request.getType()) request.setType(Type.CON);
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/CoapTcpStack.java]: please put the statement on a new line and in between curly braces

File: californium-core/src/main/java/org/eclipse/californium/core/network/stack/CoapTcpStack.java [hunk 2]
```diff
@@ -106,13 +107,15 @@ public CoapTcpStack(NetworkConfig config, Outbox outbox) {
...
 		top.sendRequest(request);
 		// CoAP over TCP does not have acknowledgements. Everything is automatically acknowledged.
 		request.setAcknowledged(true);
 	}
 
 	// delegate to top
 	@Override public void sendResponse(Exchange exchange, Response response) {
+		if (null == response.getType()) response.setType(Type.CON);
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/network/stack/CoapTcpStack.java]: please put the statement on a new line and in between curly braces

File: element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpConnectorTest.java [hunk 3]
```diff
@@ -26,6 +26,9 @@
 @RunWith(Parameterized.class)
 public class TcpConnectorTest {
 
+	private static final int NUMBER_OF_THREADS = 1;
```
Comment (sophokles73) [inline element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpConnectorTest.java]: indentation?

File: element-connector/src/test/java/org/eclipse/californium/elements/tcp/TlsConnectorTest.java [hunk 4]
```diff
@@ -35,6 +35,8 @@
 
 public class TlsConnectorTest {
 
+	private static final int NUMBER_OF_THREADS = 1;
```
Comment (sophokles73) [inline element-connector/src/test/java/org/eclipse/californium/elements/tcp/TlsConnectorTest.java]: indentation?

Comment (boaks) [inline element-connector/src/test/java/org/eclipse/californium/elements/tcp/TlsConnectorTest.java]: About the formatting: 
sorry I'm a little lost:

```
	@Override public void destroy() {
		for (Layer layer : layers)
			layer.destroy();
	}

		@Override public void receiveRequest(Exchange exchange, Request request) {
			// if there is no BlockwiseLayer we still have to set it
			if (exchange.getRequest() == null)
				exchange.setRequest(request);
			if (hasDeliverer()) {
				deliverer.deliverRequest(exchange);
			} else {
				LOGGER.severe("Top of CoAP stack has no deliverer to deliver request");
			}
		}
```

So, is it OK to correct only my changes, or is more required?

**Comment by sophokles73** [comment]
I would like to squash all of these commits into one since they are all fixing small issues with the TCP support. This will also make it easier to cherry pick it on master. Any objections?

**Comment by boaks** [comment]
OK, from my side.

**Comment by sophokles73** [comment]
merged

**Pull request closed without merging**